File: Java/src/main/java/Solution1001_1100/Solution1037.java
package Solution1001_1100;    
/**
 * @author windows
 */
public class Solution1037 {
    public boolean isBoomerang(int[][] points) {
        if ((points[0][0] - points[1][0]) == 0 && (points[1][0] - points[2][0]) != 0){
            double k1 = (points[0][1] - points[1][1]) / (points[0][0] - points[1][0]);
            double k2 = (points[1][1] - points[2][1]) / (points[1][0] - points[2][0]);
            return k1 != k2;
        }
        return true;
    }
    
    
    public static void main(String[] args) {
        Solution1037 solution1037 = new Solution1037();
    }
}

File: C++/src/Solution1001_1100/Solution1037.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution1037.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

bool isBoomerang(vector<vector<int>>& points) {
  bool isBoomerang(vector<vector<int>>& points) {
    vector<int> v1 = {points[1][0] - points[0][0], points[1][1] - points[0][1]};
    vector<int> v2 = {points[2][0] - points[0][0], points[2][1] - points[0][1]};
    return v1[0] * v2[1] - v1[1] * v2[0] != 0;
  }
}